rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/ContentModelUpdater.kt

```diff
@@ -28,34 +26,11 @@ import com.jetbrains.rider.projectView.indexing.contentModel.tryInclude
 import kotlinx.coroutines.Runnable
 import java.io.File
 
-class ContentModelUpdater(project: Project,
-    private val unityProjectDiscoverer: UnityProjectDiscoverer,
-                          private val contentModel: ContentModelUserStore): LifetimedProjectComponent(project)  {
-
-    init{
-            project.messageBus.connect().subscribe(ProjectManager.TOPIC, object : ProjectManagerListener {
-            override fun projectOpened(project: Project) {
-                if (unityProjectDiscoverer.isUnityProject) {
-
-                    val alarm = SingleAlarm(Runnable(::onRootFoldersChanged), 1000, ModalityState.any(), project)
-                    val listener = Listener(alarm)
-
-                    // Listen to add/remove of folders in the root of the project
-                    VirtualFileManager.getInstance().addAsyncFileListener(listener, project)
-
-                    // Listen for external packages that we should be indexing
-                    PackageManager.getInstance(project).addListener(listener)
-
-                    alarm.request()
-                }
-            }
-        })
-    }
-
+class ContentModelUpdater : ProjectManagerListener {
```

Comment:
This is a misuse of project listeners. ContentModelUpdater has API and stores state and even has Alarm-based events, what fits naturally here is a service with a bunch of listeners. But okay